### ISSUE TITLE
Sonarr unable to restart or shutdown from application

### ISSUE THREAD

**Issue by houndtt** [SynoCommunity/spksrc#3859]
**Describe the bug**
Of late I noticed that when I attempt to restart or shutdown Sonarr on my Synology NAS the package does not complete the desired action. For shutdowns the package does not actually stop and for restarts the package appears to stop but is unable to start again. In either case, the web interface stops responding as the action does not seem to complete.

**Logs**
Sample trace log file for a restart: https://paste.ubuntu.com/p/yjCDVtdVvQ/.

**System Information**

 - Sonarr Version: 2.0.0.5338
 - Operating System: DSM 6.2.2-24922 Update 4
 - mono (macOS/Linux) Version: 5.18.0.240

**Additional context**
For the restart problem cited above the logs (nzbdrone.log - https://paste.ubuntu.com/p/k5kmF8pNTV/) seem to indicate that the original process never stopped. As such to fix the issue temporarily, I had to manually stop the existing process as follows:
```
cd /volume1/@appstore/python/bin
sudo ./start-stop-daemon -K -q -u "sc-nzbdrone"
```

**Comment by BenjV** [comment]
Does stopping and starting from the package center work correctly?

**Comment by houndtt** [comment]
> Does stopping and starting from the package center work correctly?

Yes it does when executed by itself. If I attempt a restart for example within the application and then go into the package center, it shows the application as not running and unable to start when I click run. This is a fairly new development since I used to be able to restart from within the application without issue before.

> By the way.
> 
> > /volume1/@appstore/python/bin/start-stop-deamon
> 
> Does not exist there.

I would have to disagree since a `ls -all` in that directory shows the executable.

> start-stop-deamon is a Linux utility that is found via the environment and has nothing to do with python.
> Also Sonarr is not a python application.

I agree that it's a Linux utility and not specific to Python but I used it because a google search for stopping Sonarr from the command line refers to this method. I'm sure there are many other ways to achieve the same result.

**Comment by BenjV** [comment]
Very strange to find it in that locations, but maybe the one who created the python package distributed with python.

Did it happen after an update from Sonarr?
Most likely it is a problem from Sonarr itself, because the SynoCommunity package is not changed and stopping/starting from the package center works correct.
So you should post this issue int he Sonarr forum.

**Comment by houndtt** [comment]
It is possible that the problem occurred after an internal application update however I can't be sure as I rarely trigger a restart from within the application. I also checked using the logs but I only see one restart entry for yesterday during the past three months so I can't tell you exactly when the problem began.

I did post this exact issue to the Sonarr forum under ticket https://github.com/Sonarr/Sonarr/issues/3527 however they abruptly referred me here since they don't maintain the package. I'm not sure where to go next...

**Comment by houndtt** [comment]
> I installed Sonarr on my test Nas and experienced the same.
> Restarting Sonarr from its own interface does stop it, it but does not restart it.
> After updating Sonarr from within Sonarr it restarted fine with the new version, so clearly it is a Sonarr issue.

I'm a bit confused, are you saying that with a base install from Synocommunity (which is Sonarr v2.0.0.5301) the issue happened or didn't happen? Also when you updated to the current version within the app (which is Sonarr v2.0.0.5338) the problem stopped or did it then commence?

**Comment by BenjV** [comment]
No I tested to much at the same time.
I corrected my post.
Restarting will always work on my system but with a whole page of errors but you have to wait a few minutes before the restart happens.

Referring to here is just the easy way out for them.
It is clearly a Sonarr issue and not a issue of the Syno package.
Doing things from the user interface of Sonarr has nothing to do with the Syno Package.
Stopping Sonarr from the Package center will just kill the Sonarr process and starting it wil start the application again and that is both working fine.
The problem is the restarting and that's done by Sonarr itself.

**Comment by houndtt** [comment]
Since your results were different than mine I decided to take your approach and install a VirtualDSM on my unit using the Virtual Machine Manager. From the base install the restart and shutdown functions seemed to work:

 - Sonarr Version: 2.0.0.5301
 - mono (macOS/Linux) Version: 5.18.0.240
 - Operating System: DSM 6.2.2-24922 Update 4

Then I proceeded to install the built-in update to Sonarr and re-test. This was also successful for restart and shutdown:

 - Sonarr Version: 2.0.0.5338
 - mono (macOS/Linux) Version: 5.18.0.240
 - Operating System: DSM 6.2.2-24922 Update 4

As such now I am confused as to why my production system is not working. Perhaps I should uninstall and reinstall Sonarr? Is there a safe way to do this and retain my configurations for the PVR?

**Comment by houndtt** [comment]
Okay, I believe I've solved the issue. First I tried to backup and restore the configuration. There is a good [Backup and Restore](https://github.com/Sonarr/Sonarr/wiki/Backup-and-Restore) tutorial online. However when the configurations were restored the same problem occurred.

Doing some investigation, I came across this [post](https://forums.sonarr.tv/t/restore-from-synology/13094/2) which suggested a corrupt configuration database. I did the suggested SQLite database integrity checks however no corruption was found in the database structure. I then did a clean install in VirtualDSM and configured each screen one by one. When I configured the 'Media Management' settings using 'Advanced Settings' the problem reoccured.

I could only surmise that something in the episode formatting was off. To fix this, I turned off 'Advanced Settings' and switched the 'Separator' and 'Numbering Style' to something else then saved it. I then restarted it from the package center and proceeded to change my settings back to the previous values via the pulldown menus. Once this was done the configuration was once again saved.

Following this, the application was able to reboot from within the application UI. I then tried a shut down but this did not complete. I was however able to shut down from the package center. I tried this a second time and it then worked.
